### PR TITLE
[REVIEW] Fix hex-to-integer logic when string has prefix '0x'

### LINKED ISSUE #5113: [BUG]`hex_to_integers` fails to convert strings appended with `0x` for dtype 'int64'
**Describe the bug**
`hex_to_integers` and `stol` results doesn't match for `0xAABBCCDD` for dtype `int64`.

**Steps/Code to reproduce bug**
```
TEST_F(StringsConvertTest, HexToInteger)
{
  std::vector<const char*> h_strings{
    "1234", nullptr, "98BEEF", "1a5", "CAFE", "2face", "0xAABBCCDD", "112233445566"};
  cudf::test::strings_column_wrapper strings(
    h_strings.begin(),
    h_strings.end(),
    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));

  std::vector<int64_t> h_expected;
  for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
    if (*itr == nullptr)
      h_expected.push_back(0);
    else
      h_expected.push_back(std::stol(std::string(*itr), 0, 16));
  }

  auto strings_view = cudf::strings_column_view(strings);
  auto results      = cudf::strings::hex_to_integers(strings_view, cudf::data_type{cudf::INT64});

  std::cout<<"Result : "<<std::endl;
  cudf::test::print(results->view());
  cudf::test::fixed_width_column_wrapper<int64_t> expected(
    h_expected.begin(),
    h_expected.end(),
    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; }));
  std::cout<<"Expected : "<<std::endl;
  cudf::test::print(expected);
  cudf::test::expect_columns_equal(*results, expected);
}

```

**Expected behavior**
Results should match `stol` output

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #5113 
The conversion logic is intended to work like `std::stol()` which allows the string to optionally start with `0x`.
The code was incorrectly converting characters `[0-9][a-z][A-Z]` and should be handling `[0-9][A-F][a-f]`. This error was introduced during the port from NVStrings.
Additionally the error was not caught due to another error in the gtests.
This PR fixes both errors.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/tests/strings/integers_tests.cu [hunk 1]
```diff
@@ -164,20 +164,38 @@ TEST_F(StringsConvertTest, HexToInteger)
...
-      h_expected.push_back(0);
-    else
-      h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
+  {
+    std::vector<int32_t> h_expected;
+    for (auto itr = h_strings.begin(); itr != h_strings.end(); ++itr) {
+      if (*itr == nullptr)
+        h_expected.push_back(0);
+      else
+        h_expected.push_back((int)std::stol(std::string(*itr), 0, 16));
```
Comment (rgsl888prabhu) [inline cpp/tests/strings/integers_tests.cu:173]: Why not use stoi, may be I am unaware of something .

Comment (aschaffer) [inline cpp/tests/strings/integers_tests.cu:173]: C-style casts are doing more than they should (they are all-encompassing). Consider `static_cast<int>()` instead, which should be more efficient.

Comment (davidwendt) [reply cpp/tests/strings/integers_tests.cu:173]: The `std::stoi` throws a C++ exception when values are outside it's range and I want to just truncate the value using a cast in this case.

Comment (aschaffer) [reply cpp/tests/strings/integers_tests.cu]: You can use a `std::stringstream` to `<<` the string into and then use `>>` to extract the int/long, etc:
```
#include <iostream>
#include <string>
#include <sstream>

template<typename T>
T from_string(std::string const& str)
{
    T result;
    std::stringstream ss;
    ss<<str;
    ss>>result;
    return result;
}

int main(void)
{
    auto num_str = "12345";
    std::string str_num{num_str};
    auto value = from_string<int>(str_num);
    std::cout<<"value: "<<value<<'\n';
}
```

**Pull request merged**